### PR TITLE
Tests for splitting output files using `TimeInterval`

### PR DESCRIPTION
**Pull request by josuemtzmo**
Adds tests for splitting output based on time-interval, e.g., 

```Julia
ow = JLD2OutputWriter(model, (; u);
                      filename = "test.jld2",
                      ...
                      file_splitting = TimeInterval(3seconds)
                      )
```

### CODE REVIEW

**Comment by glwagner** [comment]
We should definitely get `TimeInterval` to work. This is important! Why doesn't it work? I don't understand the reason.

**Review by glwagner — CHANGES_REQUESTED** [review]
Let's get `TimeInterval` to work.

File: test/test_jld2_output_writer.jl [hunk 1]
```diff
@@ -88,6 +88,52 @@ function test_jld2_file_splitting(arch)
...
+    end
+    ow = JLD2OutputWriter(model, (; u=model.velocities.u);
+                          dir = ".",
+                          filename = "test.jld2",
+                          schedule = IterationInterval(1),
+                          init = fake_bc_init,
+                          including = [:grid],
+                          array_type = Array{Float64},
+                          with_halos = true,
+                          file_splitting = FileTimeSplit(3seconds),
```
Comment (glwagner) [inline test/test_jld2_output_writer.jl:107]:
```suggestion
                          file_splitting = TimeInterval(3seconds),
```

start with this change then we can see the errors

File: test/test_netcdf_output_writer.jl [hunk 2]
```diff
@@ -90,6 +90,46 @@ function test_netcdf_file_splitting(arch)
...
+    fake_attributes = Dict("fake_attribute"=>"fake_attribute")
+
+    ow = NetCDFOutputWriter(model, (; u=model.velocities.u);
+                            dir = ".",
+                            filename = "test.nc",
+                            schedule = IterationInterval(2),
+                            array_type = Array{Float64},
+                            with_halos = true,
+                            global_attributes = fake_attributes,
+                            file_splitting = FileTimeSplit(4seconds),
```
Comment (glwagner) [inline test/test_netcdf_output_writer.jl:107]:
```suggestion
                            file_splitting = TimeInterval(4seconds),
```

**Comment by josuemtzmo** [comment]
> We should definitely get `TimeInterval` to work. This is important! Why doesn't it work? I don't understand the reason.

Yes, I was also surprised. I retried and now it passed the tests, so I guess at some point I was doing something weird with `TimeInterval`, but I'm not sure what it was. It seems to work now.

**Comment by glwagner** [comment]
Awesome! And to be more clear, I think the reason we want it to work is so that we can also reuse all the other schedules. Another reason to use it is because we may need to change `TimeInterval` in the future if we support different time types (for example `DateTime`, or other time types that solve the annoying problem of rounding error). So in all those cases it will be nice not to have to worry about different implementations of time interval schedules.

File: src/OutputWriters/output_writer_utils.jl [hunk 3]
```diff
@@ -12,6 +12,12 @@ using Oceananigans.Utils: AbstractSchedule
 ##### Output writer utilities
 #####
 
+struct NoFileSplitting end
+(::NoFileSplitting)(model) = false
+Base.summary(::NoFileSplitting) = "NoFileSplitting" 
+Base.show(io::IO, nfs::NoFileSplitting) = print(io, summary(nfs))
+
+
```
Comment (navidcy) [inline src/OutputWriters/output_writer_utils.jl:20]:
```suggestion
```

File: test/test_jld2_output_writer.jl [hunk 4]
```diff
@@ -88,6 +88,52 @@ function test_jld2_file_splitting(arch)
...
+                          init = fake_bc_init,
+                          including = [:grid],
+                          array_type = Array{Float64},
+                          with_halos = true,
+                          file_splitting = TimeInterval(3seconds),
+                          overwrite_existing = true)
+
+    push!(simulation.output_writers, ow)
+
+    # 531 KiB of output will be written which should get split into 3 files.
```
Comment (navidcy) [inline test/test_jld2_output_writer.jl:112]: the comment about file size is not relevant here, right?

File: test/test_netcdf_output_writer.jl [hunk 5]
```diff
@@ -90,6 +90,46 @@ function test_netcdf_file_splitting(arch)
...
+                            schedule = IterationInterval(2),
+                            array_type = Array{Float64},
+                            with_halos = true,
+                            global_attributes = fake_attributes,
+                            file_splitting = TimeInterval(4seconds),
+                            overwrite_existing = true)
+
+    push!(simulation.output_writers, ow)
+
+    # 531 KiB of output will be written which should get split into 3 files.
```
Comment (navidcy) [inline test/test_netcdf_output_writer.jl:112]: comment about filesize is not relevant, right?

File: test/test_netcdf_output_writer.jl [hunk 6]
```diff
@@ -891,4 +932,4 @@ for arch in archs
         test_netcdf_vertically_stretched_grid_output(arch)
         test_netcdf_regular_lat_lon_grid_output(arch)
     end
-end
+end
```
Comment (navidcy) [inline test/test_netcdf_output_writer.jl:935]: we usually have an empty line at the end of file

```suggestion
end

```

**Review by navidcy — APPROVED** [review]
lgtm

**Comment by navidcy** [comment]
@josuemtzmo could you update the first PR comment to reflect the latest status?

File: test/test_jld2_output_writer.jl [hunk 7]
```diff
@@ -266,11 +312,12 @@ for arch in archs
...
-        #####
-        ##### File splitting
-        #####
+        ####
+        #### File splitting
+        ####
 
-        test_jld2_file_splitting(arch)
+        test_jld2_file_splitting_size(arch)
+        test_jld2_file_splitting_time(arch)
```
Comment (glwagner) [inline test/test_jld2_output_writer.jl:320]:
```suggestion
        test_jld2_size_file_splitting(arch)
        test_jld2_time_file_splitting(arch)
```

this doesn't matter, but does maintain our "english language style" where adjectives proceed nouns (yes, different than the way many codes are written, or French 😆 )

**Review by glwagner — APPROVED** [review]
awesome contribution! I agree with @navidcy's comments but except for the cosmetic stuff it looks good

File: test/test_netcdf_output_writer.jl (hunk 5 above)
Comment (josuemtzmo) [reply test/test_netcdf_output_writer.jl]: Yes, I deleted it

File: test/test_jld2_output_writer.jl (hunk 4 above)
Comment (josuemtzmo) [reply test/test_jld2_output_writer.jl]: Yes, I deleted it.

**Comment by navidcy** [comment]
Great! When all tests pass you can merge.

**Comment by navidcy** [comment]
excellent @josuemtzmo !

**Pull request merged**